### PR TITLE
Iterations 1 and 2

### PR DESCRIPTION
**Pull request by Abdeboskey**
Create spec files and classes for `Card`, `Turn`, `Deck`, & `Round`

### CODE REVIEW

File: lib/round.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def take_turn(guess)
+    turn = Turn.new(guess, current_card)
+    turns << turn
+    turn
+  end
+
+  def calculate_num_correct_turns(turns)
+    turns.reduce(0) { |memo, turn| turn.correct? ? memo + 1 : memo }
```
Comment (Henchworm) [inline lib/round.rb:21]: Very clever, I didn't know you could use reduce this way!

File: spec/round_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+deck = Deck.new([card_1, card_2, card_3, card_4])
+
+RSpec.describe Round do
+  before(:each) do
+    @round = Round.new(deck)
+  end
+
+  it 'exists' do
+    expect(@round).to be_instance_of(Round)
```
Comment (Henchworm) [inline spec/round_spec.rb:18]: Turing like these tests but I hate them. If you're calling Round.new, of course it's a round!

File: spec/deck_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+require './lib/card'
+require './lib/deck'
+
+card_1 = Card.new("What is my name?", "Rontron", :People)
```
Comment (Henchworm) [inline spec/deck_spec.rb]: These card variables should live below the 'Rspec.describe.' It's very useful to use 'let' blocks at the top of an Rspec file to set a variable for all tests: 
let!(:card1) { Card.new("What is my name?", "Rontron", :People) } 

https://relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let

File: spec/round_spec.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+require './lib/card'
+require './lib/turn'
+require './lib/deck'
+require './lib/round'
+
+card_1 = Card.new("What is my name?", "Rontron", :People)
```
Comment (Henchworm) [inline spec/round_spec.rb]: Same with var declaration -- put them in the Rspec describe

File: lib/turn.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    @guess = guess
+    @card = card
+  end
+
+  def correct?
+    card.answer == guess
+  end
+
+  def feedback
+    correct? ? "Correct!" : "Incorrect! The correct answer was '#{card.answer}'"
```
Comment (Henchworm) [inline lib/turn.rb:13]: Ternary operator is good for cases like this, but can become confusing -- just be sparing with using it!

**Review by Henchworm — COMMENTED** [review]
Quite good! Miles better than my mod 1 flash card project! Small comments about test setup and nitpicks

File: lib/round.rb (hunk 1 above)
Comment (Abdeboskey) [reply lib/round.rb:21]: Reduce is one of the more powerful/popular array iterators (or ENUMS) in the JS world. Very nice to find a relative in ruby land

File: spec/round_spec.rb (hunk 2 above)
Comment (Abdeboskey) [reply spec/round_spec.rb:18]: I feel you brother I feeeeeeel you.

File: spec/round_spec.rb (hunk 4 above)
Comment (Abdeboskey) [reply spec/round_spec.rb]: Encapsulation!

File: lib/turn.rb (hunk 5 above)
Comment (Abdeboskey) [reply lib/turn.rb:13]: Ternary op is another of the modern JS bread and butter tools. I'd be curious to learn more about why twould be confusing?

File: spec/deck_spec.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
+require './lib/card'
+require './lib/deck'
+
+RSpec.describe Deck do
+  let(:card_1) { Card.new("What is my name?", "Rontron", :People) }
+  let(:card_2) { Card.new("Where is the best training track?", "Aurora Reservoir", :Places) }
+  let(:card_3) { Card.new("How long is the Miami UltraSkate?", "24 hours", :SkateEvents) }
+  let(:card_4) { Card.new("Where does the A2A roadskate begin?", "Athens, GA", :SkateEvents) }
+  let(:cards) { [card_1, card_2, card_3, card_4] }
```
Comment (Abdeboskey) [inline spec/deck_spec.rb:9]: @Henchworm is this more like it?

**Pull request merged**